pkg/validating-webhooks/hive/v1/clusterpool_validating_admission_hook.go

```diff
@@ -239,6 +243,10 @@ func (a *ClusterPoolValidatingAdmissionHook) validateUpdate(admissionSpec *admis
 
 	allErrs = append(allErrs, validateClusterPlatform(specPath, newObject.Spec.Platform)...)
 
+	if newObject.Spec.Inventory != nil {
+		allErrs = append(allErrs, validateInventory(specPath, newObject.Spec.Inventory)...)
```

Comment:
Hm. The purpose of this validation was to bounce on:
```yaml
spec:
  inventory: []
```
But since `Spec.Inventory` is not a pointer type, I don't know whether a CR with the `inventory` field omitted will still unmarshal to effectively an empty list. Have you tried this live with those two different cases?

On that note, we do have unit test suites for webhooks -- eventually please be sure to add tests for the new checks. (Though I'm not sure whether you would be able to test the `[]`-vs-`nil` thing in that context.)

Reply:
> Have you tried this live with those two different cases?
no

> eventually please be sure to add tests for the new checks.

I will try

Reply:
When I used `inventory: []` or `inventory: null`, kubernetes just omitted the lines and the verification never failed.  Can I remove the validation of `spec.inventory` in such case?

Reply:
Yeah, that’s fine. If the inventory field was a pointer type it might work, but it’s not worth changing. Don’t forget to update the design accordingly.